src/storage/src/hummock/store/version.rs

```diff
@@ -91,23 +115,27 @@ pub struct StagingVersion {
 impl StagingVersion {
     pub fn prune_overlap<'a>(
         &'a self,
-        epoch: HummockEpoch,
+        min_epoch: HummockEpoch,
+        max_epoch: HummockEpoch,
         table_id: TableId,
         key_range: &'a (Bound<Vec<u8>>, Bound<Vec<u8>>),
     ) -> (
         impl Iterator<Item = &ImmutableMemtable> + 'a,
         impl Iterator<Item = &SstableInfo> + 'a,
     ) {
         let overlapped_imms = self.imm.iter().filter(move |imm| {
-            imm.epoch() <= epoch
+            imm.epoch() <= max_epoch
+                && imm.epoch() > min_epoch
                 && range_overlap(key_range, imm.start_user_key(), imm.end_user_key())
         });
 
+        // TODO: Remove duplicate sst based on sst id
```

Comment:
why will we have duplicate SST?

Reply:
I think it is because we are checking staging `ssts` in all ReadVersion?

Reply:
In this PR， Batch Query will obtain staging ssts cross the read_versions, so it may contains some duplicated staging ssts , need to dedup.